corehq/warehouse/tests/test_integration.py

```diff
@@ -84,31 +100,67 @@ def tearDownClass(cls):
 
         FormProcessorTestUtils.delete_all_sql_forms(cls.domain)
 
-        FormStagingTable.clear_records()
-        FormFact.clear_records()
-        DomainStagingTable.clear_records()
-        DomainDim.clear_records()
-        UserStagingTable.clear_records()
-        UserDim.clear_records()
-        super(FormFactIntegrationTest, cls).tearDownClass()
-
-    def test_loading_form_fact(self):
+        FormStagingLoader.clear_records()
+        FormFactLoader.clear_records()
+        DomainStagingLoader.clear_records()
+        DomainDimLoader.clear_records()
+        UserStagingLoader.clear_records()
+        UserDimLoader.clear_records()
+        SyncLogStagingLoader.clear_records()
+        SyncLogFactLoader.clear_records()
+        AppStatusFormStagingLoader.clear_records()
+        AppStatusSynclogStagingLoader.clear_records()
+        ApplicationStatusFactLoader.clear_records()
+        super(AppStatusIntegrationTest, cls).tearDownClass()
+
+    def test_loading_app_stats_fact(self):
         batch = self.batch
 
-        DomainStagingTable.commit(batch)
-        self.assertEqual(DomainStagingTable.objects.count(), len(self.domain_records))
-
-        DomainDim.commit(batch)
-        self.assertEqual(DomainDim.objects.count(), len(self.domain_records))
-
-        UserStagingTable.commit(batch)
-        self.assertEqual(UserStagingTable.objects.count(), len(self.user_records))
-
-        UserDim.commit(batch)
-        self.assertEqual(UserDim.objects.count(), len(self.user_records))
-
-        FormStagingTable.commit(batch)
-        self.assertEqual(FormStagingTable.objects.count(), len(self.form_records))
-
-        FormFact.commit(batch)
-        self.assertEqual(FormFact.objects.count(), len(self.form_records))
+        seen = set()
+
+        def commit_with_dependencies(clazz, path=None):
```

Comment:
not sure if `clazz` is a nice name here. Can use `_class` or `dependency_or_loader_class` or `for_class` or something better?